File: frontend/src/services/dashboard.service.ts
import api from './api';

export interface DashboardStats {
    monthlyRevenue: number;
    monthlyOrders: number;
    totalProducts: number;
    totalUsers: number;
}

export interface DailyOrderStat {
    date: string;
    count: number;
}

export interface LowStockProduct {
    productId: string;
    productName: string;
    productVariantId: string;
    size: string | null;
    color: string | null;
    stockQuantity: number;
    imageUrl?: string;
    price: number;
    categoryName?: string;
}

export const dashboardService = {
    async getStats(): Promise<DashboardStats | null> {
        try {
            const response = await api.get('/admin/dashboard/stats');
            return response.data.result;
        } catch (error) {
            console.error('Fetch dashboard stats failed', error);
            return null;
        }
    },

    async getDailyOrders(startDate: string, endDate: string): Promise<DailyOrderStat[]> {
        try {
            const response = await api.get(`/admin/dashboard/daily-orders?startDate=${startDate}&endDate=${endDate}`);
            return response.data.result || [];
        } catch (error) {
            console.error('Fetch daily orders failed', error);
            return [];
        }
    },

    async getLowStockProducts(): Promise<LowStockProduct[]> {
        try {
            const response = await api.get('/admin/dashboard/low-stock');
            return response.data.result || [];
        } catch (error) {
            console.error('Fetch low stock products failed', error);
            return [];
        }
    }
};

File: backend/src/main/java/com/zola/services/dashboard/DashboardService.java
package com.zola.services.dashboard;

import com.zola.dto.response.dashboard.DashboardResponse;

import java.time.LocalDateTime;
import java.util.List;
import com.zola.dto.response.dashboard.DailyOrderStat;
import com.zola.dto.response.dashboard.LowStockProduct;

public interface DashboardService {
    DashboardResponse getStats();
    List<DailyOrderStat> getDailyOrders(LocalDateTime startDate, LocalDateTime endDate);
    List<LowStockProduct> getLowStockProducts();
}
